src/auth/negotiate/kerberos/negotiate_kerberos_pac.cc

```diff
@@ -362,6 +365,154 @@ getextrasids(char *ad_groups, uint32_t ExtraSids, uint32_t SidCount)
     return ad_groups;
 }
 
+static char *
+get_resource_group_domain_sid(uint32_t ResourceGroupDomainSid)
+{
+
+    if (ResourceGroupDomainSid != 0) {
+        uint8_t rev;
+        uint64_t idauth;
+        char dli[256];
+        char *sid;
+        int l;
+
+        align(4);
+
+        // ResourceGroupDomainSid structure:
+        //  4 bytes nauth
+        //  1 byte revision  = 1
+        //  1 byte nsub (it is equal to the number of dashes minus two)
+        //  6 bytes idauth   (for NT Authority it is 5)
+        //  4 bytes sauth1
+        //  ... nauth timss
+        //  4 bytes sauthN
+
+        uint32_t nauth = get4byt();
+
+        // check if nauth math will produce invalid length values on 32-bit
+        static uint32_t maxGidCount = (UINT32_MAX - 4 - 1 - 1 - 6)/4;
+        if (nauth > maxGidCount) {
+            debug((char *) "%s| %s: ERROR: Too many subAuths in the ResourceGroupDomainSID: nauth = %d > %d\n",
+                  LogTime(), PROGRAM, nauth, maxGidCount);
+            return nullptr;
+        }
+
+        // length = revision[1byte]+nsub[1byte]+idauth[6bytes]+nauth*sauth[4bytes]
+        size_t length = 1 + 1 + 6 + nauth*4;
+
+        sid=(char *)xcalloc((length + sizeof(uint32_t))*sizeof(char), 1); // +sizeof(uint32_t) for storing length of a SID
```

Comment:
FYI, I have omitted the `*sizeof(char)` from first parameter of these calls to `xcalloc()` because the second parameter is where the array object size is supposed to be passed.
 ... and `xcalloc(N, 1)` is already just shorthand for `xcalloc(N, sizeof(char))`